src/hint_processor/builtin_hint_processor/uint384_extension.rs

```diff
@@ -0,0 +1,352 @@
+            d5: vm.get_integer((addr + 5)?).map_err(|_| {
+                HintError::IdentifierHasNoMember(name.to_string(), "d5".to_string())
+            })?,
+        })
+    }
+
+    pub(crate) fn from_var_name<'a>(
+        name: &str,
+        vm: &'a VirtualMachine,
+        ids_data: &HashMap<String, HintReference>,
+        ap_tracking: &ApTracking,
+    ) -> Result<Uint768<'a>, HintError> {
+        let base_addr = get_relocatable_from_var_name(name, vm, ids_data, ap_tracking)?;
+        Uint768::from_base_addr(base_addr, name, vm)
+    }
+}
+
+fn pack_extended(num: Uint768, num_bits_shift: usize) -> BigUint {
+    let limbs = [num.d0, num.d1, num.d2, num.d3, num.d4, num.d5];
+    #[allow(deprecated)]
+    limbs
+        .into_iter()
+        .enumerate()
+        .map(|(idx, value)| value.to_biguint().shl(idx * num_bits_shift))
+        .sum()
+}
+
+/* Implements Hint:
+       %{
+           def split(num: int, num_bits_shift: int, length: int):
+               a = []
+               for _ in range(length):
+                   a.append( num & ((1 << num_bits_shift) - 1) )
+                   num = num >> num_bits_shift
+               return tuple(a)
+
+           def pack(z, num_bits_shift: int) -> int:
+               limbs = (z.d0, z.d1, z.d2)
+               return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
+
+           def pack_extended(z, num_bits_shift: int) -> int:
+               limbs = (z.d0, z.d1, z.d2, z.d3, z.d4, z.d5)
+               return sum(limb << (num_bits_shift * i) for i, limb in enumerate(limbs))
+
+           a = pack_extended(ids.a, num_bits_shift = 128)
+           div = pack(ids.div, num_bits_shift = 128)
+
+           quotient, remainder = divmod(a, div)
+
+           quotient_split = split(quotient, num_bits_shift=128, length=6)
+
+           ids.quotient.d0 = quotient_split[0]
+           ids.quotient.d1 = quotient_split[1]
+           ids.quotient.d2 = quotient_split[2]
+           ids.quotient.d3 = quotient_split[3]
+           ids.quotient.d4 = quotient_split[4]
+           ids.quotient.d5 = quotient_split[5]
+
+           remainder_split = split(remainder, num_bits_shift=128, length=3)
+           ids.remainder.d0 = remainder_split[0]
+           ids.remainder.d1 = remainder_split[1]
+           ids.remainder.d2 = remainder_split[2]
+       %}
+*/
+pub fn unsigned_div_rem_uint768_by_uint384(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let a = pack_extended(Uint768::from_var_name("a", vm, ids_data, ap_tracking)?, 128);
+    let div = pack(
+        BigInt3::from_var_name("div", vm, ids_data, ap_tracking)?,
+        128,
+    );
+    let quotient_addr = get_relocatable_from_var_name("quotient", vm, ids_data, ap_tracking)?;
+    let remainder_addr = get_relocatable_from_var_name("remainder", vm, ids_data, ap_tracking)?;
+    let (quotient, remainder) = a.div_mod_floor(&div);
```

Comment:
Is there any chance `div` is `0`? If there is, we should catch it, I believe.

Reply:
Added check to hint & previous uint384 hints with similar behaviour

Reply:
Edit: I added it to another PR that depends on this, let me copy it to this branch